features/reset/reset_and_delete/feature.go

```diff
@@ -127,18 +125,24 @@ func CheckResult(ctx context.Context, r *harness.Runner, run client.WorkflowRun)
 	if next.GetEventType() != enums.EVENT_TYPE_WORKFLOW_EXECUTION_TERMINATED {
 		return errors.New("expected last event to be terminated")
 	}
-	// Ensure original run is findable via visibility & has correct status
-	resp, err := r.Client.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
-		Namespace: r.Namespace,
-		Query:     "WorkflowId = '" + origRun.GetID() + "'",
-	})
-	if err != nil {
-		return err
-	}
-	r.Require.Equal(1, len(resp.GetExecutions()))
-	r.Require.Equal(enums.WORKFLOW_EXECUTION_STATUS_TERMINATED, resp.GetExecutions()[0].Status)
-
-	return err
+	// Use eventually since visibility is eventually consistent.
+	r.Require.EventuallyWithT(func(t *assert.CollectT) {
+		// Ensure original run is findable via visibility & has correct status
+		resp, err := r.Client.ListWorkflow(ctx, &workflowservice.ListWorkflowExecutionsRequest{
+			Namespace: r.Namespace,
+			Query:     "WorkflowId = '" + origRun.GetID() + "'",
+		})
+		assert.NoError(t, err)
+		if err != nil {
+			return
+		}
+		assert.Len(t, resp.GetExecutions(), 1)
+		if len(resp.GetExecutions()) != 1 {
+			return
```

Comment:
Why do we check this again and why is it OK to return without error here?

Reply:
Oh I think I see -- if it's `!= 1` then the previous `assert.Len` failed and this tick is not clean, and we don't want a panic on the next line. (Would be nice to have a cleaner way!)

Reply:
Yeah, EventuallyWithT is notoriously hard to use. `assert` expressions don't break out of the function and `require` breaks the retry loop. So you're stuck with evaluating certain conditions twice.